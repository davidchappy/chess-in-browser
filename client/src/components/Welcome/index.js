import React from 'react';
import './Welcome.css';
import Header from '../shared/header';

const Welcome = (props) => {
  return (
    <div className="Welcome">
      <Header {...props} welcome={true} />
      <p className="Welcome-intro">
        <button onClick={() => props.createGuestGame("Alex", "Lucas")}>Start Game As Guest</button>
      </p>
    </div>
  )
}

export default Welcome;