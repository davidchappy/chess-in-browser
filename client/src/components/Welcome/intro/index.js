import React from 'react';
import './intro.css';

const Intro = (props) => {
  return (
    <p className="Welcome-intro">
      <button onClick={() => props.createGuestGame()}>Start Game As Guest</button>
    </p>
  )
};

export default Intro;

