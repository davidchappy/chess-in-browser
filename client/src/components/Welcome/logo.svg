<?xml version="1.0" encoding="UTF-8"?>
<svg width="181px" height="281px" viewBox="0 0 181 281" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <g id="knight" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <image id="chess_pieces_black-copy-3" x="0" y="0" width="181" height="281" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAALUAAAEZCAYAAADR+mSIAAAABGdBTUEAA1teXP8meAAAEI5JREFUeAHtnU12nLwShu17spKMM8s+vFjvIzOPvZVcF6H8qUE0EtSv9PocBxBCKr31UF0StPP6Mu/PX8Ghvwq2haZuKjCTMx4gfnt765Lu4+OjWv/z85PLZ9KSxxxyO7ojvkHuhZi8dQTy1pMr2KNruR122OMfYS27btgtkKnbVpg3JlK/AHsjisfhSE5YYOaI/P7+vujJxy3iXoR5aRrRukVhmzojROoHmEm2EuhyfyvpHYi3beE4jgKZI/UOZpaVQC4j9BZsDZgRqVl9/23WSP23hPZMRqpLYNPvz58/z6rjfHIFMkLdBTT7h28CjtqAm5UZb/u/ZEO6BHSZbjDcxfpyMglg7pkCmaBuApoiMYNbDp7AZrjpPEVqgF0qNM5+hvTjcEJ4xQ0MNl3LYCMVuaJk3GuiQ90UnUneEtZS7qNyrgOwWYlxtlHTD4rOt4FudROD3Vof9WIrEDFSN8NM0p5F4lb5GWykIq2Kxa0XLVLfApomfrVJYo/8VyeP682w5P89/aGuvALRoG4eoVSELjtElC7VyLsfCermKK0BNLuQ0xA+7tkiWveopVc3CtQhgC5lRhpSqpFrPwrUoVS7m4YgYvu6MwLUYlGaYOJ3OyRkvRqtqe8CbEweJZzR0YY31GJAd4y5qerdaE2dFG0A7CbVZSp5Qt0MtMxQr7VyJ1pTjwR2EbWvGYGruhTwhLrZUM3VjmdGcKS9Czb1AbCfKS17zgtqsyj969evW4oB7FvyuVzsBXXzYHujNEEoOVkkQwF2s7tCVPSA2ixKSypMYDPcd9td28Hk8a6QB9d7QH1gyr64N0rvW4hbArD1fGMNtUmUJmCkUxANFwBsDVVfXiyh7gJaMkrfnSzqSP+vVYAtr64l1PLWD9IiwJZ1pBXUXVFadog5WgPYcn6ygrrLYonUgyDJkFeXwhRgY2WkFKZz3wJqROkOpxDYBdwdV6IqK2ABNffVtJWI0rWOIk8Wa/YC7JoqbWXaUCNKt/mhWgtgV2U5LdSEGkCfyn9eAWCfa7StoQn1tq/TY+nU4+43y08NNqoAsPuEDgV1n+nntcubpNw/vzJeDYDd7hOtP7renXpkh65d8ns113e7tfx2z7ggVw8dqYNoLGoGIva5nBpQd0fpczNRo1QAYJdq7Pc1oN73clKC1ONEoMppgF0RZS0KAfWxeTjzTIECbDxWL4SSnnBcTj0QrQuvXNgtvhws7dML1vhegkjtq79Y72vUFmsvc0OSUF+O0pkFjGR7kY5EMsvcFkmozY1Hh3sFALbt17n2HlhLkE8fSnPpxOxgR/zvMS45coSLisne6XDOcmg6/9UerYpMN3EE1Kf46FdgmP/8/t3U2e8/f5b/AxJg1+WSyqkxSazre1pKQBPMrUBTgz11Z0xFpKA+dd5RhZnzaY7QR9pIlc8GtjvUUo7L1g4D3RN174xxJrCRU98h5cm1DO2TKl1pxLN2Ws8R2F928SP1YSeQgLqViCf1WgDeXm4Vobf9rhGb4R4SbEC99fqFYwal51JawaAfT7jXqD0c2K459cyTRL4RGO6eG0Kq7moDpyNSzbq34wq1++idDWCwe82QvBFGBBtQ9xIlXJ+g6oGU6169IWrmjwY2cuqal43LSrCPcmyGmUyTBJqHSm2OkmMDavaq85ZBLeEtTeLzZZn0/ihgS818Lz8mn3myKA2lVHvrEqUUG1JmNbeDnLpZqnkqrp8KaVdFJKC+HKXnwSTtSFOCLQH1ZY8h9bgsnfqFFK2zRmxXqNU9gw5uK5ARbEB92+3jN1CAnSIdcYU621/3Hx/f4xGuYB9XCHTGFepAOsCUBgWKiN1Q268KoPbTPmXPBdhhUxFAnRItX6OLVCQk2IDal4+0vRPYRdQONQ5AHcod+YwpwA4TtQF1Po7CWbyJ2u5wu0ONZb1wjF42aI3adD2B7QY3Xj297EJcWFOgAJvfz+ZqZm/9AWqWHFtxBQ4AV4fbPf0QVxINhlTAMu8G1CERGNeoMnprjRJQaymLdg8VWMFWm0iGgBorIIf+H/aEJtgSUL++v78PKz4Glk8BCajzjRoWD62A1PKKyPcU8fWucVh79kczy8mixjfXsU49DkdhRsJAl/CycXSOfmvnuM7dLdKPuwri+qoCR9ByOYNfvfhmYSiosQpy05sBLm+Jwtpgh4I6gE9ggpECDPZXd1Lzum/LAfW3FNgZRQFAPYonMY5vBQD1txTYkVLg2SSwPLemIOKPy7GkJ+VJtLMowLnyFt7yWFsqQK2t8KTtM9w0fAa6LNOURXLmKfJUkQaLJ4uaLo/V9gq8JIcvyKlj+Xg6azTy6pBQ4yHMdGyLDjgk1KIjRGPTKRAWakTr6VgUG3BYqGmEAFvMz1M1hCU9Z3fzctfWDKvlr22/IxxLRmp8rauTiAJoWtIqf5e1XTpf1Olsfd7qklCrqDhqClLAWlujLQF/ZbiLa1S0HqVRpB85PPkN/hfY3+9KIEWpOy98pK6bnb/0BpAcxZGiHGAgDTXy6gOhFYoZ7mVbpiizpynSUCv4Dkt7jaKWkE8dxTWgRrRupFDjvYe166kB14C60aV91UZdBelT4VLtHeCXWlG6aE2VyEaxnzRQi404ZkPfKxrK5i2Ac/6t3Jdb81pQIwVpdOmNVZDGHqrVlsjoPaHU6l8L6qqSKKwroJhb1zv8VxoC7C9TRFMPGhqg/udg939nA1sjl2YnakItnoJMMlm0yq+ZAfFIyQ0fbTWBpj41oT4aE8oPFCjya2uwh3pxKh3Uo0drJ7DFojVFYf6t3btrlK6dEisTG8wTi8S+Zc59zPBtc+2PaNZys/1b3FSbU22Hhd3PPm1UubOI1OK5dZu8uWs5TRwlRSNwj34l+9m1ZQH1rtO7BaOnIKyPA9jLu9vcf9ZtSqizin3D7mcf5TearV9qkffWe5YpBdQyOqq1UuS4VmAv+W5msK2gFs+rZ0lB6G4pwFa7eTYNX5rIRbkRrKAmzcTB3jhi6MNE+fWlG0LSeZZQS9o9ZVsOYDc9lKEIvUZpd6AJDECd7PYowLbIsb/z66PUoigPATS5E98mTwY1mUtgFzBpj4Bh/fukT66jbUtT+x7GiD5hnOHp4pEnV8g8fHhkUohypB8h3HDNiCIVudbAoFelh3qmpb0agwB7r4oH1Fja2/vhVgnAfpTPA+pHC3AEBYQVANTCgno1h2j9n/JeUCMF+c8HYnsA+5+UXlCLORINPSoAsPFE8ZGIQY5mBxuRehCQt8OYGWxPqJFXb0nUObZ4R0TH8outekJ90eT9ZbM/gNkr8q9kjdZ0MBXYQ0B95FSUP3zBYBqwAfUE5FPEninH9oYaefUEN5X1EL2hth7v1P3NEq0B9WSYzwA2oJ4M6mK4w04cAXXh5Vl2R584RoAak0Xfu2m4iB0Bal+XTtz7ml+TAkOBPQzUeKp47e4cEexhoL7mUlxFChRgDyEIoB7CjfcHsYI9RBoSBWpMFu9zebuFUcCOAvVth6ABGQVGABtQy7CAVgIpAKgDOSOKKdmjNaCOQlIwOzKDDaiDwRTJnKxgA+pIFAW0JSPYgDogSEFNSrOGPQzUM/+dau2bYI3W1E0KsIeBWtuxs7dfgB1eCkAd3kVxDMySXwPqOMyksCQD2IA6BUqxjIwONqCOxUsaayKDDajTYBTW0HArIkNAjeU8H+DXaO3T+ZNeh4D6yfhwSlmBIg0JE7EBtbLTZ2g+WsQG1DNQZzDGImIb9Pa8C0D9XB+c7VAgCthRoL78/5VjkthB3SRVo0A9idzjDzNCtE4NNaJ06JvEbTUkNdShXTqxcd6rIRGgvpxPT8xN+KF7piERoL7kIKQel2Sb4qKUUAPoHGx6RWtvqLtTDwCdA+iNlaaTRm+oN2PH4WgKeEwaPaFGlB6N4CDjeXWyA0A7Ce/V7efnJ3VtwptnpG7WF3l0s1SoaHXnbJTuitIAeqNe4kOraB06UgPoxAQ7mm4NdXOUBtCOVOh2rb68Z5K4f2m0DOTt7a1JLgDdJFPKShYpyA8DZZqjM9kCoA08MngXmlB3f8wA6MFpMxqeVvqBdMPIgRm70U5BpCM1YM5I2WA2S65+AOjB4Mg6HNFI3bK6gbw5Kyp57JaC+nSFAzDngSK7pVJQV3UAyFVZUKisgGRO/WAqgH6QAweGCqhBbTgGdAUFHhRQgRpR+kFjHBgroAK18RjQHRR4UEAcakTpB31x4KCAKNQA2sGD6HKngBjUAHqnLQqcFBCD2sl+dAsFdgpIQf26vnm16wAFUMBaASmore1Gf1DgUAFJqBGtD2XGCUsFJKEmuwG2pffQV1UBaaipE4BdlRqFpIDF3EsDarIdYJMK+DlSQOtrhEt/WlBT4wD7yKUoV1VAE2oyHGCrui9X4xapBymiDXUu1WGthQKqqQcNwAJqRGsLVIL3YRWlraAOLjfMM1RAPUoDakNvztyVZZQG1DOTZj92kygNqO0dO12Pa5Q2A9oSakwWp8PZ5ulhTVaL1Y9avyibRwHTKE2ymkJtPWGYh5t4I/VIO1gFS6jN71geJLa2CngCTSP1AG3566ge/xOqrWvn7M0baC+o2dt/ATZLkX9bpJYegfJBQG8DELUf3JHzIEJ0LpXzhpptAdysRKJtpOhcyhYFarYJcLMSwbdRgSbZokHNrkS+zUoE2hYgk1VR2Ylr2JdoS9Rmn2JSyUrYbjcgU+dhYWZlwhu4GvoAOBsP0FkJ2W1GkEsFLB++lP327tPNt/utiN/bLuofK1DqfVwr4JksUNekQ95dUwVltu9+QG8oYKFA5kiN11kVCBkhpcsMtYJL0eSqQJYFhKrDUhu/jgi5ddW1/YVrlE7PBCJ1v+9xRXAFRoAauXVwyKzNGwFqa83QX3AFAHVwB8G8fgVGgLr6CL1fClwxigIjQP2Cd0BGwVFmHENALSMFWlkVSP/Jl31NEmvUwvdi8UQxLRuZI3X6iCLMo0hzI6RymaFGLi2C8b6RFey0QSPrRwzSjj2L4iVZH5tnhBpAi+N73GBGsDNBvXwcjpDzHSMU80y2yWMWqBGdnXnPBHYGqAG0M9DcfRawI0ONdINpCraNnmdHhRrRORjIW3MiR+2IUAPoLUFBj6OCHQlqpBtB4X1mVkSwo0CN6PyMnATnIsHtDTWicwJgW02MArYn1H/f3t5ePj4+WjVDvSQKrHC7seXV8QI0+whgsxLjbD2jtsdbeg9Aj+NGjKRUgF5nKN72W9LM8rzmviXUNDAArenNgG0X7+qYgW2VfhzCjNQjIIkKJlmmIxZQA2gFSDI2aQW2NtQAOiN9ijZbgK0JNYBWhCNz09pga0ENoDNTZ2C7Jtgaqx8A2gCK7F0UqyLiQ5GG+nDZBqsc4r5L32Cxji06FmmoX+jRN36ggKcC4lDXBoMoXVMFZVoKmECtZTzaHUaBw7T1yggB9RXVcI2YAhoTRuklPdE7Tkw5NJRBATEWpRpaYK5NEpFPZ+DJ10bpNesfN4dzCPPNdnH5RApwCvIFN3/S3wq2Vy9ughlReiIyBYd6N3L3RuommAXHh6YmVOBu5G6N1N0wI0pPSKPSkHsj91mk7oZZaVxodmIFeiP3UaS+BTOi9MQEGgz9LHJvI/UtmA3Ggy6gAH+h9+VotYQjtRjMiNKgzlqBbeQmqBegrQ1Bf1BAS4H/A+Fzg4RwNo3vAAAAAElFTkSuQmCC"></image>
    </g>
</svg>