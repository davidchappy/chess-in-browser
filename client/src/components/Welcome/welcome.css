.Welcome {
  text-align: center;
}

.Welcome-intro {
  font-size: large;
}
