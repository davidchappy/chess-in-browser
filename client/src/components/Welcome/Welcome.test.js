import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import Welcome from './index';

describe('rendering Welcome (Snapshot Test)', () => {
  const component = renderer.create(<Welcome />);
  const json = component.toJSON();

  it('renders without crashing', () => {
    expect(json).toMatchSnapshot();
  });
});

describe('rendering Welcome (mount test)', () => {
  const welcomeWrapper = mount(
    <Welcome />
  );

  it('renders header text', () => {
    const h2 = welcomeWrapper.find('.Welcome-header h2');
    expect(h2.text()).toBe('Welcome to Chess');
  });

  it('renders a button to start a guest game', () => {
    const button = welcomeWrapper.find('.Welcome-intro button');
    expect(button.text()).toBe('Start Game As Guest');
  });
});

describe('starting Guest Game', () => {
  const createGuestGame = jest.fn();
  const welcomeWrapper = mount(
    <Welcome createGuestGame={createGuestGame}/>
  )

  it('calls createGuestGame', () => {
    const button = welcomeWrapper.find('.Welcome-intro button');
    button.simulate('click');
    expect(createGuestGame).toBeCalled();
  });
});