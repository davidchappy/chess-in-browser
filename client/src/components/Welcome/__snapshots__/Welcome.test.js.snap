// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering Welcome (Snapshot Test) renders without crashing 1`] = `
<div
  className="Welcome"
>
  <div
    className="Welcome-header"
  >
    <img
      alt="logo"
      className="Welcome-logo loaded"
      src="logo.svg"
    />
    <h2>
      Welcome to Chess
    </h2>
  </div>
  <p
    className="Welcome-intro"
  >
    <button
      onClick={[Function]}
    >
      Start Game As Guest
    </button>
  </p>
</div>
`;
